### PR TITLE
Use HTTP header for app authentication

### PR DESCRIPTION
**Pull request by levb**
#### Summary
Use `AuthStyleInHeader` to make client_id/client_secret to be passed in the HTTP header.

image

https://github.com/golang/oauth2/blob/2e8d9340160224d36fd555eaf8837240a7e239a7/internal/token.go#L174

#### Ticket Link
#431, https://mattermost.atlassian.net/browse/MM-34646

### CODE REVIEW

**Review by catalintomai — APPROVED** [review]

File: webapp/package-lock.json [hunk 1]
```diff
@@ -7341,7 +7341,8 @@
       "version": "1.3.0",
       "resolved": "https://registry.npmjs.org/growly/-/growly-1.3.0.tgz",
       "integrity": "sha1-8QdIy+dq+WS3yWyTxrzCivEgwIE=",
-      "dev": true
+      "dev": true,
```
Comment (hanzei) [inline webapp/package-lock.json:7344]: Unrelated change?

**Review by hanzei — APPROVED** [review]
LGTM :+1: 

Thanks for looking into this @levb

**Review by iomodo — APPROVED** [review]

**Comment by DHaussermann** [comment]
Understood. @levb or @hanzei How urgent is this? I'm curious is the plan is to cut a release that is v.`2.0.0` + the 2 PRs or following the regular process of creating a release from `master`.

**Comment by hanzei** [comment]
@DHaussermann Yes, there will be a patch release based on `v2.0.0` containing the two PRs.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Located a server with oAuth app using deprecated method and upgraded
- Upgrade appears to be seamless - connected users remained connected on upgrade
- Tested connect, connect private, and disconnect
- Tested on desktop and browser
- Tested multiple unique users connecting as a sanity check
- Verified after an hour no emails are received regarding the use deprecated oAuth method
- No changes were required to my oAuth application on GitHub side
- Spot checked functionality such as issue creation and subscription delivery to ensure APIs are working

LGTM!
